scripts/ci/pre_commit_check_order_setup.py

```diff
@@ -0,0 +1,175 @@
+        dependent = pattern_dependent.findall(sub_dependent)
+
+        src = [pattern_dependent_version.sub('', p) for p in dependent]
+        res = _check_list_sorted(src, f"Order of sub-dependencies group: {group_name}") and res
+    return res
+
+
+def check_alias_dependent_group(setup_context: str) -> bool:
+    """
+    Test for an order of each dependencies groups declare like
+    `alias_dependent_group = dependent_group_1 + ... + dependent_group_n` in setup.py
+    """
+    pattern = re.compile('^\\w+ = (\\w+ \\+.*)', re.MULTILINE)
+    dependents = pattern.findall(setup_context)
+
+    res = True
+    for dependent in dependents:
+        src = dependent.split(' + ')
+        res = _check_list_sorted(src, f"Order of alias dependencies group: {dependent}") and res
+    return res
+
+
+def check_install_and_setup_requires(setup_context: str) -> bool:
+    """
+    Test for an order of dependencies in function do_setup section
+    install_requires and setup_requires in setup.py
+    """
+    pattern_install_and_setup_requires = re.compile(
+        '(setup_requires) ?= ?\\[(.*?)\\]', re.DOTALL)
+    install_and_setup_requires = pattern_install_and_setup_requires.findall(setup_context)
+
+    res = True
+    for dependent_requires in install_and_setup_requires:
+        pattern_dependent = re.compile('\'(.*?)\'')
+        dependent = pattern_dependent.findall(dependent_requires[1])
+        pattern_dependent_version = re.compile('[~|><=;].*')
+
+        src = [pattern_dependent_version.sub('', p) for p in dependent]
+        res = _check_list_sorted(src,
+                                 f"Order of dependencies in do_setup section: "
+                                 f"{dependent_requires[0]}") and res
+    return res
+
+
+def check_extras_require(setup_context: str) -> bool:
+    """
+    Test for an order of dependencies in function do_setup section
+    extras_require in setup.py
+    """
+    pattern_extras_requires = re.compile(
+        r'EXTRAS_REQUIREMENTS: Dict\[str, Iterable\[str\]] = {(.*?)}', re.DOTALL)
+    extras_requires = pattern_extras_requires.findall(setup_context)[0]
+
+    pattern_dependent = re.compile('\'(.*?)\'')
+    src = pattern_dependent.findall(extras_requires)
+    return _check_list_sorted(src, "Order of dependencies in: extras_require")
+
+
+def check_provider_requirements(setup_context: str) -> bool:
+    """
+    Test for an order of dependencies in function do_setup section
+    providers_require in setup.py
+    """
+    pattern_extras_requires = re.compile(
+        r'PROVIDERS_REQUIREMENTS: Dict\[str, Iterable\[str\]\] = {(.*?)}', re.DOTALL)
+    extras_requires = pattern_extras_requires.findall(setup_context)[0]
+
+    pattern_dependent = re.compile('"(.*?)"')
+    src = pattern_dependent.findall(extras_requires)
+    return _check_list_sorted(src, "Order of dependencies in: providers_require")
+
+
+if __name__ == '__main__':
+    setup_context_main = setup()
+    result = True
```

Comment:
Is it not enough to check whether the errors variable contains elements?

Reply:
I think we could just use `error` to check the error elements too.

Reply:
And others LGTM

Reply:
Yeah I added it later (initially I printed errors as they were found). Indeed. I can simplify it :)

Reply:
Fixed